zephyr/CMakeLists.txt

```diff
@@ -711,6 +711,9 @@ zephyr_library_sources_ifdef(CONFIG_COMP_GOOGLE_HOTWORD_DETECT
 zephyr_library_sources_ifdef(CONFIG_DTS_CODEC
 	${SOF_AUDIO_PATH}/module_adapter/module/dts/dts.c
 )
+if (CONFIG_DTS_CODEC)
+zephyr_library_import(DtsCodec ${SOF_AUDIO_PATH}/module_adapter/lib/release/libdts-sof-interface-i32.a)
```

Comment:
zephyr_library_sources_ifdef so you don't need external if

Reply:
Do you mean I could remove "if (CONFIG_DTS_CODEC)" and keep zephyr_library_import() only?

Reply:
@cujomalainey I don't think zephyr_library_sources_ifdef can work since this is to import external libraries

Reply:
Yes, zephyr_library_sources_ifdef doesn't work to import library. I also tested it again to confirm it can't build pass. This leads to the question in my last comment.

Reply:
Yeah, zephyr's cmake integration is elaborate and has lots of great hidden features like implicit ifdef/kconfig integration (also lots of hidden "you must use this" traps), but it's not as orthogonal as you'd want.  No one ever added a `zephyr_library_import_ifdef()`.

And in fact this function itself is a little obscure, and I had to look it up myself.  It's not actually used in-tree, even Zephyr's "external library" sample code doesn't use it, and makes do with cmake-native "add_library()/target_link_libraries()" code.  But it certainly looks like it does the right thing.  It just lacks kconfig integration. 

(That was way too much text.  This looks fine to me basically)

Reply:
One nitpick might be the file location though: stuffing it down under src/audio seems weird?  Aren't we trying to put binary artifacts in /third_party?  Not the expert. Don't trust my review.

Reply:
yea i think i bad lookup on the existence of a function, apologies for the noise

Reply:
ditto to @andyross please import from third party folder

Reply:
@joechengxperi , why there is no directory and file with this path? module_adapter/lib/release/libdts-sof-interface-i32.a, or you are adding in a separate PR?

Reply:
@btian1 It's DTS proprietary library. I can't submit it here. Currently, it's only shared to the partners who need to test DTS functionality or Google for Chromebook project release.

Reply:
@cujomalainey I wonder if I could move the import to another PR. 
1. This is more like code reorganization rather than functionality.
2. I noticed that several google header files are under third_party folder. I'm thinking if it's necessary to move DTS header files here as well.
3. There is also one suggestion from @btian1  (https://github.com/thesofproject/sof/pull/8379#pullrequestreview-1698605677) to move DTS code to src/audio/
I think it might be better to put them together in another thread to discuss. Let me know how you think.

Reply:
+1 for moving the DTS library header files and proprietary libraries (locally) to third_party/. So in the future one wouldn't expect the existence while seeing the library is imported from the path of third_party.

As for moving DTS code to src/audio/, it makes sense but I don't think we should do it now. Two reasons:
1) module API migration is a long-term project, the related optimization task cab be put off unless it blocks some thing.
2) it will be difficult to cherry-pick commits to production branch in the future once you move codes

Reply:
> it will be difficult to cherry-pick commits to production branch in the future once you move codes

As long as you don't change the code too much while moving it, git is actually pretty good at cherry-picking changes across moves and renames. Easy enough to try.

Example with `sof/rimage` -> `sof/tools/rimage`  https://github.com/thesofproject/rimage/commit/b38fef772bf5ae6de2610ff5ff6e3a983a817f2